File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int check(TreeNode root){
        if(root==null) return 0;

        int lh=0,rh=0;
        if(root.left!=null) lh=check(root.left);
        if(root.right!=null) rh=check(root.right);

        return 1+Math.max(lh,rh);
    }
    public int maxDepth(TreeNode root) {
        return check(root);  
    }
}

File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """

        def check(node):
            if node == None:
                return 0
            
            lh = check(node.left)
            rh = check(node.right)

            return 1 + max(lh, rh)
        return check(root)
        